src/middleware/IllnessReasonFeatureMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+
+import { loggerInstance } from 'app/middleware/Logger';
+import { Feature } from 'app/utils/Feature';
+import { isFeatureEnabled } from 'app/utils/FeatureChecker';
+import { ENTRY_PAGE_URI } from 'app/utils/Paths';
+
+@provide(IllnessReasonFeatureMiddleware)
+export class IllnessReasonFeatureMiddleware extends BaseMiddleware {
+
+    public handler(req: Request, res: Response, next: NextFunction): void {
+
+        if (isFeatureEnabled(Feature.ILLNESS_REASON)) {
```

Comment:
Feels like this logic is duplicated in at least one more place. Wouldn't it be better top have one configurable middleware that is tested once and just configured in places it is needed?

Reply:
Good idea - i've created a configurable middleware, see `FeatureToggleMiddleware.ts`